client/src/components/overview/ProductOverview.jsx

```diff
@@ -88,25 +121,35 @@ class ProductOverview extends React.Component {
     }
   }
 
-  updateDefaultStyle() {
+  updateDefaultStyle(styles) {
     console.log('Updating default style');
-    const defaultStyle = this.getDefaultStyle();
+    const defaultStyle = this.getDefaultStyle(styles);
     this.updateState(defaultStyle);
   }
 
-  getDefaultStyle() {
-    for (const style of this.styles) {
+  updateRatingsProperties(ratings) {
+    console.log(`Product Ratings: ${JSON.stringify(ratings)}`);
+    const numberOfReviews = this.getNumberOfReviews(ratings);
+    this.setState({
+      numberOfReviews: numberOfReviews,
+      averageRating: this.getAverageRating(ratings, numberOfReviews)
```

Comment:
Why call `getAverageRating` in `setState` but assigning a temp variable for `getNumberOfRatings`?

Reply:
It seemed like I was courting problems by using the value of one state to determine another while they are being set together. Cleaner to just calculate it outside of setting state.